blueocean-dashboard/src/main/js/components/RunDetails.jsx

```diff
@@ -137,10 +138,13 @@ class RunDetails extends Component {
     afterClose = () => {
         const { router, params } = this.context;
         if (this.opener) {
-            router.goBack();
+            // step back in the history to the item that's prior to initial load of RunDetails
+            const offsetToInitialRoute = this.initialHistoryLength - history.length;
+            router.go(offsetToInitialRoute - 1);
```

Comment:
What API exactly are you thinking of calling? You can't read from the browser history. We could create our own history array in the `LocationService`; I'm not opposed to that.